### PR TITLE
Fix webpack publicPath Windows issue

### LINKED ISSUE #262: Wepback Path issue on Windows
@chaelli's description copied from https://github.com/vigetlabs/gulp-starter/pull/257#issuecomment-202039278

> since this update I get a 404 on the http://localhost:3000/javascripts/app.js - there's also no "javascripts" folder in the "public" folder. 
Any idea what could be wrong? the paths seem to be correct...
 
**Update:**     
> ouch.. found the issue.. Windows again ;)
just add ".replace(/\/g, '/')" to the following line in webpack-multi-config.js to fix it:
var publicPath = path.join(config.tasks.js.dest, '/')

### PR DESCRIPTION
**Pull request by greypants**
Resolves #262 

Also changes some minor naming related to #265.

@chaelli if you get a chance, would you mind checking out that branch and seeing if all's well?

### CODE REVIEW

**Comment by chaelli** [comment]
Hi @greypants 
sadly this doesn't work... I did some debugging:
gutil.log(arguments)
  gutil.log(path.join.apply(arguments))
  gutil.log(path.join.apply(arguments).replace(/\\/g, '/'))

writes:
[19:49:26] { '0': 'javascripts', '1': '/' }
[19:49:26] .
[19:49:26] .
maybe just join first and then expect a string in the pathToUrl method.. replace.. return.

**Comment by greypants** [comment]
Ack. I'm a dummy. The `apply` signature is `fun.apply(thisArg, [argsArray])`. I forgot to pass context (this). Not sure how this slipped by... I feel like this should have broken the build.

Alright... I need to stop being sloppy and writing new code without writing tests. I'll fix this and add some JS testing to ensure the expected values are getting returned, and that the build DOES break if the tests don't pass.

Thanks for testing that for me! Will fix.

**Comment by greypants** [comment]
Fixed and added Mocha tests that get run on Travis! @chaelli let me know if those tests cover all the variations. All green :white_check_mark: !

File: gulpfile.js/lib/pathToUrl.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+var path = require('path')
+
+module.exports = function pathToUrl() {
+  // Normalizes Windows file paths to valid url paths
+  return path.normalize(path.join.apply(this, arguments).replace(/\\/g, '/'))
```
Comment (chaelli) [inline gulpfile.js/lib/pathToUrl.js]: return path.normalize(path.join.apply(this, arguments)).replace(/\\/g, '/') => replace after normalize

Comment (chaelli) [inline gulpfile.js/lib/pathToUrl.js]: after that it works

Comment (greypants) [inline gulpfile.js/lib/pathToUrl.js]: Thanks. Can you paste an example file path I can add to the tests?

Comment (greypants) [inline gulpfile.js/lib/pathToUrl.js]: OOOOooohhhh. right. Normalize uses the system slashes (which undos the string replace). I was thinking it just combined double slashes.

**Comment by greypants** [comment]
Done. Also realized `path.join` also normalizes. Didn't need to do that explicitly. Thanks again!

**Pull request merged**